# IMAP Essential Toolkit

Welcome to the IMAP Essential Toolkit repository! This toolkit is designed to provide you with a curated list of essential tools and software necessary to kickstart your journey into microbiome data analysis. Whether you're a seasoned researcher or just diving into the world of microbiome analysis, this repository aims to guide you through the installation process for the essential tools you'll need.

## Getting Started

To get started with the IMAP Essential Toolkit, follow these steps:

1. **Review the Tools List:** Take a look at the list of essential tools provided in this repository.

2. **Install the Tools:** Detailed installation instructions for each tool can be found on the [IMAP Essential Toolkit GitHub Pages](https://your-username.github.io/imap-essential-toolkit/). Navigate to the respective tool's page for installation guidance.

3. **Note on Updates:** Please be aware that the tools included in this repository may be subject to updates by their developers. If you encounter any issues with the installation process or if the provided instructions do not work, we recommend checking the respective tool's official website for the latest information and updates.

## Contributing

If you have suggestions for additional tools to include in the IMAP Essential Toolkit or if you encounter any issues with the installation process, feel free to open an issue or submit a pull request.

So, let's get started and unlock the potential of your microbiome datasets with the IMAP Essential Toolkit!


<br>

# Related Works
## IMAP GitHub Repos

| IMAP-Repo                                      | Description                                               | GH-Pages                                             |
| ---------------------------------------------- | --------------------------------------------------------- | ----------------------------------------------------- |
| [OVERVIEW](https://github.com/tmbuza/imap-project-overview/) | IMAP project overview                                     | [Link](https://tmbuza.github.io/imap-project-overview/) |
| [PART 01](https://github.com/tmbuza/imap-essential-software/) | Software requirement for microbiome data analysis with Snakemake workflows | [Link](https://tmbuza.github.io/imap-essential-software/) |
| [PART 02](https://github.com/tmbuza/imap-sample-metadata/) | Downloading and exploring microbiome sample metadata from SRA Database | [Link](https://tmbuza.github.io/imap-sample-metadata/) |
| [PART 03](https://github.com/tmbuza/imap-download-sra-reads/) | Downloading and filtering microbiome sequencing data from SRA database | [Link](https://tmbuza.github.io/imap-download-sra-reads/) |
| [PART 04](https://github.com/tmbuza/imap-read-quality-control/) | Quality Control of Microbiome Next Generation Sequencing Reads | [Link](https://tmbuza.github.io/imap-read-quality-control/) |
| [PART 05](https://github.com/tmbuza/imap-bioinformatics-mothur/) | Microbial profiling using MOTHUR and Snakemake workflows | [Link](https://tmbuza.github.io/imap-mothur-bioinformatics/) |
| [PART 06](https://github.com/tmbuza/imap-bioinformatics-qiime2/) | Microbial profiling using QIIME2 and Snakemake workflows | [Link](https://tmbuza.github.io/imap-qiime2-bioinformatics/) |
| [PART 07](https://github.com/tmbuza/imap-data-processing/) | Processing Output from 16S-Based microbiome bioinformatics pipelines | [Link](https://tmbuza.github.io/imap-data-preparation/) |
| [PART 08](https://github.com/tmbuza/imap-exploratory-analysis/) | Exploratory Analysis of 16S-Based Microbiome Processed Data | [Link](https://tmbuza.github.io/imap-data-exploration/) |


## Session information

For a detailed overview of the tools and versions suitable for this guide, explore the [session information](session_info.txt).

## Citation
> Please consider citing the [iMAP article](https://rdcu.be/b5iVj) if you find any part of the iMAP practical user guides helpful in your microbiome data analysis.

Buza, T. M., Tonui, T., Stomeo, F., Tiambo, C., Katani, R., Schilling, M., â€¦ Kapur, V. (2019). iMAP: An integrated bioinformatics and visualization pipeline for microbiome data analysis. BMC Bioinformatics, 20. https://doi.org/10.1186/S12859-019-2965-4

## :tada: Raise awareness
> Please help increase awareness of freely available tools for microbiome data analysis.
See [Dimensions of the iMAP article](https://badge.dimensions.ai/details/id/pub.1117740326)
